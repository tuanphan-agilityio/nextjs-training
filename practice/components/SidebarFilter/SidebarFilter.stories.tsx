import { Story, Meta } from '@storybook/react';

import SidebarFilter from '.';

export default {
  title: 'SidebarFilter',
  component: SidebarFilter,
} as Meta;

const Template: Story = (args) => <SidebarFilter {...args} />;

export const Default = Template.bind({});
Default.args = {};
