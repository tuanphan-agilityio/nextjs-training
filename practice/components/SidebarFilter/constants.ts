const FILTER_CATEGORIES = ['Men', 'Women', 'Children'];

export { FILTER_CATEGORIES };
