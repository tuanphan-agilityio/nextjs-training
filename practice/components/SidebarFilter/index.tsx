import { FC, memo } from 'react';
import clsx from 'clsx';

import FilterIcon from '../../public/icons/filter.svg';

import { FILTER_CATEGORIES } from './constants';

const SidebarFilter: FC = () => {
  return (
    <div>
      <div className='flex items-center text-xl'>
        <p className='font-primary-regular leading-4'>Filter</p>
        <FilterIcon className='ml-2' />
      </div>
      <div className='mt-10 text-md'>
        <p className='mb-3'>Categories</p>
        <ul className='flex gap-6 pb-6 border-b-2 border-solid border-tertiary'>
          {FILTER_CATEGORIES.map((category, index) => (
            <li
              key={category}
              className={clsx([
                'px-2 py-1',
                index === 0
                  ? 'bg-quaternary font-primary-bold text-primary border'
                  : 'text-secondary border border-tertiary hover:cursor-pointer',
              ])}
            >
              {category}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default memo(SidebarFilter);
