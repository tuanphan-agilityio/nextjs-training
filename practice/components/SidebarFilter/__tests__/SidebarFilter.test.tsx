import { render } from '@testing-library/react';

import SidebarFilter from '..';

describe('SidebarFilter Component', () => {
  it('renders correctly', () => {
    const { container } = render(<SidebarFilter />);
    expect(container).toMatchSnapshot();
  });

  it('displays the "Filter" text', () => {
    const { getByText } = render(<SidebarFilter />);
    const filterText = getByText('Filter');
    expect(filterText).toBeInTheDocument();
  });

  it('displays the "Categories" text', () => {
    const { getByText } = render(<SidebarFilter />);
    const categoriesText = getByText('Categories');
    expect(categoriesText).toBeInTheDocument();
  });
});
