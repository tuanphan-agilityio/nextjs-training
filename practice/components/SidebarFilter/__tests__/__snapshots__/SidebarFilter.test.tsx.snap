// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarFilter Component renders correctly 1`] = `
<div>
  <div>
    <div
      class="flex items-center text-xl"
    >
      <p
        class="font-primary-regular leading-4"
      >
        Filter
      </p>
      <svg
        class="ml-2"
        fill="none"
        height="24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          stroke="#0D1222"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-miterlimit="10"
          stroke-width="1.2"
        >
          <path
            d="M15.38 7.4a1.9 1.9 0 1 0 0-3.8 1.9 1.9 0 0 0 0 3.8ZM17.27 5.5h3.17M2.6 5.5h10.88M7.9 13.76a1.9 1.9 0 1 0 0-3.8 1.9 1.9 0 0 0 0 3.8ZM9.8 11.86h10.64M2.6 11.86h3.41M15.38 20.12a1.9 1.9 0 1 0 0-3.8 1.9 1.9 0 0 0 0 3.8ZM17.27 18.22h3.17M2.6 18.22h10.88"
          />
        </g>
      </svg>
    </div>
    <div
      class="mt-10 text-md"
    >
      <p
        class="mb-3"
      >
        Categories
      </p>
      <ul
        class="flex gap-6 pb-6 border-b-2 border-solid border-tertiary"
      >
        <li
          class="px-2 py-1 bg-quaternary font-primary-bold text-primary border"
        >
          Men
        </li>
        <li
          class="px-2 py-1 text-secondary border border-tertiary hover:cursor-pointer"
        >
          Women
        </li>
        <li
          class="px-2 py-1 text-secondary border border-tertiary hover:cursor-pointer"
        >
          Children
        </li>
      </ul>
    </div>
  </div>
</div>
`;
